File: leetcode/Pascal's Triangle.cpp
#include<vector>

class Solution {
public:
    vector<vector<int> > generate(int numRows) {
        // Note: The Solution object is instantiated only once and is reused by each test case.
        vector<vector<int> > res;
        vector<int> first(1, 1);
        vector<int> second(2, 1);
        
        if (numRows == 0) return res;
        res.push_back(first);
        if (numRows == 1) return res;
        res.push_back(second);
        if (numRows == 2) return res;
        
        for (int i = 3; i <= numRows; i++) {
            vector<int> now(i, 1);
            for (int j = 1; j < i-1; j++)
                now[j] = res[i-2][j-1] + res[i-2][j];
            res.push_back(now);
        }
        
        return res;
    }
};

File: leetcode/Pascal's Triangle.java
public class Solution {
    public ArrayList<ArrayList<Integer>> generate(int numRows) {
    	ArrayList<ArrayList<Integer>> list = new ArrayList<ArrayList<Integer>>();
    	for (int i = 0; i < numRows; i++) {
    		ArrayList<Integer> innerList = new ArrayList<Integer>();
    		innerList.add(1);
    		for (int j = 1; j < i; j++) {
    			innerList.add(list.get(i - 1).get(j) + list.get(i - 1).get(j - 1));
    		}
    		if (i > 0)
    			innerList.add(1);
    		list.add(innerList);
    	}
    	return list;
    }
}